File: src/main/java/com/theendercore/hydra/util/CommandRegistry.java
package com.theendercore.hydra.util;

import com.mojang.brigadier.tree.LiteralCommandNode;
import net.fabricmc.fabric.api.client.command.v2.ClientCommandManager;
import net.fabricmc.fabric.api.client.command.v2.ClientCommandRegistrationCallback;
import net.fabricmc.fabric.api.client.command.v2.FabricClientCommandSource;

public class CommandRegistry {

    public static void init() {
        ClientCommandRegistrationCallback.EVENT.register((dispatcher, registryAccess) -> {

            LiteralCommandNode<FabricClientCommandSource> hydraNode = ClientCommandManager.literal("hydra").build();
            LiteralCommandNode<FabricClientCommandSource> enableNode = ClientCommandManager.literal("enable").executes(HydraCommands::enable).build();
            LiteralCommandNode<FabricClientCommandSource> disableNode = ClientCommandManager.literal("disable").executes(HydraCommands::disable).build();
            LiteralCommandNode<FabricClientCommandSource> testNode = ClientCommandManager.literal("test").executes(HydraCommands::test).build();

            dispatcher.getRoot().addChild(hydraNode);
            hydraNode.addChild(enableNode);
            hydraNode.addChild(disableNode);
            hydraNode.addChild(testNode);
        });
    }
}

File: src/main/kotlin/com/theendercore/hydra/util/CommandRegistry.kt
package com.theendercore.hydra.util

import com.mojang.brigadier.CommandDispatcher
import com.mojang.brigadier.context.CommandContext
import net.fabricmc.fabric.api.client.command.v2.ClientCommandManager
import net.fabricmc.fabric.api.client.command.v2.ClientCommandRegistrationCallback
import net.fabricmc.fabric.api.client.command.v2.FabricClientCommandSource
import net.minecraft.command.CommandRegistryAccess

object CommandRegistry {
    fun init() {
        ClientCommandRegistrationCallback.EVENT.register(ClientCommandRegistrationCallback { dispatcher, _ ->
            val hydraNode = ClientCommandManager.literal("hydra").build()
            val enableNode = ClientCommandManager.literal("enable")
                .executes { HydraCommands.enable(it) }
                .build()
            val disableNode = ClientCommandManager.literal("disable")
                .executes { HydraCommands.disable(it) }
                .build()
            val testNode = ClientCommandManager.literal("test")
                .executes { HydraCommands.test(it) }
                .build()
            dispatcher.root.addChild(hydraNode)
            hydraNode.addChild(enableNode)
            hydraNode.addChild(disableNode)
            hydraNode.addChild(testNode)
        })
    }
}